scripts/golinter.sh

```diff
@@ -62,8 +62,9 @@ if [ -n "$OUTPUT" ]; then
 fi
 
 echo "Checking with go vet"
+GOTAGS=WITH_PDO_CRYPTO
 PRINTFUNCS="Debug,Debugf,Print,Printf,Info,Infof,Warning,Warningf,Error,Errorf,Critical,Criticalf,Sprint,Sprintf,Log,Logf,Panic,Panicf,Fatal,Fatalf,Notice,Noticef,Wrap,Wrapf,WithMessage"
-OUTPUT="$(go vet -all -printfuncs "$PRINTFUNCS" $1/...)"
+OUTPUT="$(go vet -all -tags "$GOTAGS" -printfuncs "$PRINTFUNCS" $1/...)"
```

Comment:
This uses the specified tag everywhere, doesn't it?

Also, I think it works only because the code vetted with the build tag is a super set of the code vetted without the build tag.
So, if anybody implements a Go file with `// +build !WITH_PDO_CRYPTO`, this is left out.

Should we maybe duplicate the `go vet` (i.e., run w/ and w/o gotags)?

Reply:
good point! yes, here we tell go vet to use `WITH_PDO_CRYTPO` tag everywhere.

Without the tag, all the files with `// +build WITH_PDO_CRYPTO` are excluded from vetting. This also causes go vet to throw an error for ERCC, since the implementation for [NewVerifier](https://github.com/hyperledger/fabric-private-chaincode/blob/8edde2f06061dfa5088aab0a7b7665708029c05d/ercc/attestation/verifier.go#L23) is missing. As ERCC is currently relying on PDO crypto, it must be built with the `WITH_PDO_CRYPTO` tag nevertheless. Having a ERCC without PDO crypto would be an alternative, but requires to write the verification code in go as well. Maybe a TODO for the future to also give SDK users the ability to verify the attestations. Technically, within ERCC there is no technical need to use the `WITH_PDO_CRYPTO` tag, in contrast to the Client SDK related components, however, I think this tag is a good indicator that this files should only be compiled if PDO crypto is available on the platform.

Ideally we could tell go vet to use the `WITH_PDO_CRYPTO` tag only in the scope of ERCC. Not sure how to do that easily. But if we do that, the PDO-based CSP implementation is also left out from vetting. This becomes complex very quickly. I suggest we see the `WITH_PDO_CRYPTO` tag not as a toggle. That is, you are not supposed to write go files using `!WITH_PDO_CRYPTO` in order to use a different implementation for something.

Alternatively, we could run go vet per component to better respect individual build options. This may require a bigger refactoring of the `golinter.sh`. I don't think it is worth the efforts at the moment, and it does not harm to run go vet just with the `WITH_PDO_CRYPTO` flag. Do you have any idea for a lightweight solution?

It seems that gotags and platform specific code in general add this complexity when using go vet. See [here](https://stackoverflow.com/questions/63745232/how-to-run-go-vet-on-all-code-regardless-of-platform).

Reply:
> Alternatively, we could run go vet per component to better respect individual build options. This may require a bigger refactoring of the golinter.sh. I don't think it is worth the efforts at the moment, and it does not harm to run go vet just with the WITH_PDO_CRYPTO flag. Do you have any idea for a lightweight solution?

Right, the right approach seem to be per-component. Not worth it now.
I though the lightweight approach was the "double" execution, but unfortunately it does not work.
It's ok as it is for now.